app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -199,6 +262,195 @@ fun HomeScreen(
   }
 }
 
+@Composable
+fun SearchBarAndFilter(
+    searchQuery: String,
+    onSearchQueryChange: (String) -> Unit,
+    onSearchButtonClick: () -> Unit,
+    showFilterPopUp: Boolean,
+    setShowFilterPopUp: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+    ) {
+        // Search bar
+        TextField(
+            value = searchQuery,
+            // function onSearchQueryChange() needs to be created in the VM
+            onValueChange = { onSearchQueryChange(it) },
+            modifier = Modifier.weight(1f),
+            maxLines = 1,
+            shape = RoundedCornerShape(32.dp),
+            colors = TextFieldDefaults.colors(
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                disabledIndicatorColor = Color.Transparent
+            ),
+            placeholder = { Text("Search event...") },
+            trailingIcon = {
+                IconButton(
+                    onClick = onSearchButtonClick,
+                    modifier = Modifier
+                        .padding(horizontal = 4.dp)
+                ) {
+                    Icon(Icons.Default.Search, contentDescription = null)
+                }
+            }
+        )
+
+        // Filter button
+        Button(
+            // function setShowFilterPopUp() needs to be created in the VM
+            onClick = { setShowFilterPopUp(!showFilterPopUp) },
+            modifier = Modifier.padding(start = 8.dp)
+        ) {
+            Text("Filter")
+        }
+    }
+}
+
+@Composable
+fun FilterPopUp(
+    onRadiusChange: (Double) -> Unit,
+    onFreeSelectionChange: (Boolean) -> Unit,
+    onCategorySelectionChange: (List<EventCategory>) -> Unit,
+    modifier: Modifier = Modifier,
+    uiState: EventsOverviewUiState = EventsOverviewUiState()
+    ) {
+    Box(
+        modifier = modifier
+    ) {
+        Card(
+            modifier = Modifier.padding(12.dp),
+            elevation = cardElevation(defaultElevation = 4.dp),
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(12.dp)
+            ) {
+                // Text input for radius
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.Start
+                ) {
+                    Text(
+                        text = "Radius: ",
+                        style = TextStyle(
+                            fontSize = 16.sp,
+                        )
+                    )
+                    TextField(
```

Comment:
nit: change the keyboard options to use just a number keyboard

Reply:
Also make sure to reformat your code to be in line with KTFMT.

Reply:
done